core/embed/trezorhal/hardware_version.c

```
@@ -1,13 +1,44 @@
 
-const char* const HW_VER_str[] = {
-    ENUM_NAME_ARRAY_ITEM(HW_VER_INVALID), ENUM_NAME_ARRAY_ITEM(HW_VER_UNKNOWN),
-    ENUM_NAME_ARRAY_ITEM(HW_VER_LEGACY),  ENUM_NAME_ARRAY_ITEM(HW_VER_1P3A),
-    ENUM_NAME_ARRAY_ITEM(HW_VER_3P0),     ENUM_NAME_ARRAY_ITEM(HW_VER_3P0A),
-};
+// since the HW_VER_t enum was designed to represent ADC values directly for
+// easy usage we cannot use regular "array of strings" method to offer name
+// reflection because the array will have gaps between items, which wasting
+// memory
+
+char const* hw_ver_to_str(HW_VER_t hw_ver) {
```

cppcheck (2.10-2)

[style] The function 'hw_ver_to_str' is never used.

(unusedFunction)